### PR TITLE
Encode language change for "Espanol" button

### LINKED ISSUE #3975: "Espanol" button on patient experience should be marked as Spanish-language
## Area or Component
Patient experience. (self-registration and results)

## Quick Summary
The button to switch to the Spanish version of the patient experience is not itself styled as Spanish

## Accessibility Violation
The change in language for a portion of content is not coded. https://axe.deque.com/rule-help/en/lang-change-not-marked

In practice, this will make the text unreadable for screenreader users (imagine pronouncing "Espanol" with an American accent - E- span-oll or similar)

## Action Requested
Style the button to be `lang=es`

## Source Code (optional)
Probably the [LanguageToggler](https://github.com/CDCgov/prime-simplereport/blob/c3e6b2dc330021e425e01657eaee613c51f9e11c/frontend/src/patientApp/LanguageToggler.tsx#L4-L4)

## Original Deque Issue:
lang-change-not-marked | The change in language for a portion of content is not coded. | The "Espanol" button is not labelled as Spanish language. In practice, this means that screenreaders will read with an English accent. We should also check to ensure the Spanish translations are language-labelled correctly | Change in language is not marked | https://axe.deque.com/rule-help/en/lang-change-not-marked | serious | Yes | No | Structure | ["body > div#root:nth-of-type(1) > div.bg-base-lightest.minh-viewport:nth-of-type(1) > div.bg-white:nth-of-type(1) > header.header.border-bottom.border-base-lighter:nth-of-type(1) > div.display-flex.flex-align-center.maxw-tablet.grid-container.patient-header:nth-of-type(1) > div.display-flex.flex-align-end:nth-of-type(2) > button.usa-button.usa-button--unstyled:nth-of-type(1)"] | The change in language for a portion of content is not coded. |   | wcag2aa,wcag312a | rnv7@cdc.gov | Patient Self-Registration - Form | https://stg.simplereport.gov/app/register/9C7BB |   | 2022-07-13T22:16:55.472Z
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --

### LINKED ISSUE #4087: Language change not marked on print modal
## Area or Component
Results page - print modal

## Quick Summary
There's a button at the top of the modal for "Espanol", to get the results in Spanish. This language change isn't encoded into the application, so screenreaders will read "Espanol" with an English/American pronunciation.

## Accessibility Violation
Change in language is not marked: https://cdc-devtools.dequecloud.com/rule-help/en/lang-change-not-marked

## Action Requested
Code "Espanol" button as Spanish-language.

## Screenshots (optional)
Screen Shot 2022-07-26 at 4 29 33 PM

## Original Deque Issue
https://docs.google.com/spreadsheets/d/1b6xjubAN5Uw11DV4_B5zvSu7HeDI6Z-DpV4DMcToqAY/edit#gid=1978534823&range=26:26

### PR DESCRIPTION
**Pull request by emmastephenson**
# FRONTEND PULL REQUEST

## Related Issue

Fixes #3975 
Fixes #4087 
Fixes untracked issue where the overall document language didn't change when a user elected to view the result in Spanish.

## Changes Proposed

- Add a language change to the button specifically (so that the button text "Espanol" is marked Spanish-language)
- Changes the document `lang` attribute to reflect the language chosen in the toggler

## Additional Information

- `document.setAttribute` hack grabbed from the [i18next docs](https://github.com/i18next/react-i18next/issues/925#issuecomment-522299751) but if there's a better way to do this I'm all ears

## Screenshots

https://user-images.githubusercontent.com/80282552/182261908-60ba857d-8f4e-4a9b-8400-7d06b39b42ca.mp4

## Testing

- Validate that document `lang` attribute is correct in all locations

### Design
- [ ] Any UI/UX changes have a designer as a reviewer, and changes have been approved
- [ ] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke-tested by both the engineering and design teams

### Content
- [ ] Any content changes have been approved by content team

### Support
- [ ] Any changes that might generate new support requests have been flagged to the support team
- [ ] Any changes to support infrastructure have been demo'd to support team

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

### Documentation
- [ ] Any changes to the startup configuration have been documented in the README

### CODE REVIEW

**Review by emyl3 — DISMISSED** [review]
This is an edge case and not sure if you want to handle it in this PR but to repro:

On the SimpleReport side I am noticing the wrong language for this scenario:
- go to the "Results" page
- click "Print result" for a test result
- click "Español" in the print modal
- then close the print modal
- everything on the "Results" page will be in English but it will have `<html lang="es">`

**Comment by nathancrtr** [comment]
> This is an edge case and not sure if you want to handle it in this PR but to repro:
> 
> On the SimpleReport side I am noticing the wrong language for this scenario:
> 
> * go to the "Results" page
> * click "Print result" for a test result
> * click "Español" in the print modal
> * then close the print modal
> * everything on the "Results" page will be in English but it will have `<html lang="es">`

Good catch! This basic issue of "how can we reliably bail out of an i18n-ized state" was never really addressed at its root. In the very first implementation, following your replication steps above and then navigating to the person form would leave the _form itself_ in Spanish! We added [a hook](https://github.com/CDCgov/prime-simplereport/blob/main/frontend/src/app/patients/Components/PersonForm.tsx#L115-L121) to the person form to reset the language in such cases, but as we see here more subtle problems still persist.

**Comment by emmastephenson** [comment]
> 

@nathancrtr and @emyl3 

Interesting, thanks for the additional context Nathan! And good catch Elisa.

What I did to try and get around this was force the document language back to English on modal close. This modal isn't shared directly with patient experience, so I think forcing it back to English on close is appropriate since we don't support Spanish on the internal app. But let me know if you think there's a better implementation!

**Review by emyl3 — CHANGES_REQUESTED** [review]
Left one comment for the results page. Pxp looked good to me!

File: frontend/src/app/testResults/TestResultPrintModal.tsx [hunk 1]
```diff
@@ -217,7 +218,10 @@ export const DetachedTestResultPrintModal = ({
       <Button
         variant="unstyled"
         label={t("testResult.close")}
-        onClick={closeModal}
+        onClick={() => {
+          closeModal();
+          setDocumentLanguage("en");
```
[the remark this answers is not in the document]
Comment (emmastephenson) [reply frontend/src/app/testResults/TestResultPrintModal.tsx]: Oooh good catch. I moved both calls to the utils file and renamed that function, so now when the language is changed the document language is also automatically changed. Thanks for the through review on this Elisa!!

**Review by fzhao99 — APPROVED** [review]

**Review by emyl3 — APPROVED** [review]
Thank you! LGTM! 😹
